server/src/test/java/org/elasticsearch/snapshots/SnapshotsServiceTests.java

```diff
@@ -598,10 +971,17 @@ public void start(ClusterState initialState) {
                     @Override
                     public void connectToNodes(DiscoveryNodes discoveryNodes) {
                         // override this method as it does blocking calls
+                        boolean callSuper = true;
                         for (final DiscoveryNode node : discoveryNodes) {
-                            transportService.connectToNode(node);
+                            try {
+                                transportService.connectToNode(node);
+                            } catch (Exception e) {
+                                callSuper = false;
```

Comment:
We have to bail here if a connect fails, otherwise will block while trying to reconnect.